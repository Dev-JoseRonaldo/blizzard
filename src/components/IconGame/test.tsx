import { render, screen, waitFor } from '@testing-library/react'

import { IconGame } from '.'

const { getByRole, getByText, queryByText } = screen

const data = {
  iconUrl: 'https://res.cloudinary.com/icon.png',
  alt: 'Alternative Text',
  width: 174,
  height: 149,
}

describe('<IconGame />', () => {
  it('should render by default', async () => {
    render(<IconGame {...data} />)

    const name = queryByText('name')
    const icon = getByRole('img', { name: 'Alternative Text' })

    expect(name).not.toBeInTheDocument()
    expect(icon).toBeInTheDocument()

    await waitFor(() => {
      expect(icon).toHaveAttribute(
        'src',
        '/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Ficon.png&w=384&q=75',
      )
    })

    expect(icon).toHaveAttribute('alt', 'Alternative Text')
    expect(icon).toHaveStyle({
      width: 174,
      height: 149,
    })
  })

  it('should render with text', async () => {
    render(<IconGame {...data} name="name" />)

    const name = getByText('name')

    expect(name).toBeInTheDocument()
  })

  it('Should match snapshot', () => {
    const { container } = render(<IconGame {...data} />)

    expect(container).toMatchSnapshot()
  })
})
