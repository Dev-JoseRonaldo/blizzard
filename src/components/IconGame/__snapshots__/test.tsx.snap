// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconGame /> Should match snapshot 1`] = `
<div>
  <figure
    class="group flex flex-col items-center justify-center gap-[0.5375rem] undefined"
  >
    <img
      alt="Alternative Text"
      class=""
      data-nimg="1"
      decoding="async"
      height="149"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Ficon.png&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Ficon.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Ficon.png&w=384&q=75 2x"
      style="color: transparent;"
      width="174"
    />
  </figure>
</div>
`;
